app/preprints/detail/template.hbs

```diff
@@ -17,7 +17,8 @@
                             data-test-edit-preprint-button
                             data-analytics-name='Edit preprint button'
                             local-class='btn btn-primary'
-                            @href={{concat '/' this.model.preprint.id '/edit'}}
+                            @route='preprints.edit'
+                            @models={{array this.provider.id this.model.preprint.id}}
```

Comment:
Does `this.provider` work here? For a route template, `this.property` will typically look for properties defined in the route's `controller.ts`, and I don't see `provider` defined there. Should this provider model be coming from `this.model`?

Reply:
Ah, good catch, let me fix this